src/installer/java/baritone/install/BaritoneInstaller.java

```diff
@@ -0,0 +1,190 @@
+        });
+
+        c.gridwidth = GridBagConstraints.REMAINDER;
+        Label empty = new Label("");
+        gridbag.setConstraints(empty, c);
+        add(empty);
+
+        c.weightx = 1.0D;
+        c.gridwidth = GridBagConstraints.BOTH;
+        empty = new Label("");
+        gridbag.setConstraints(empty, c);
+        add(empty);
+
+
+        c.weightx = 1.0D;
+        c.gridwidth = GridBagConstraints.BOTH;
+        Button cancel = new Button("Cancel");
+        gridbag.setConstraints(cancel, c);
+        add(cancel);
+        cancel.addActionListener(e -> System.exit(0));
+
+        addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                System.exit(0);
+            }
+        });
+
+        c.gridwidth = GridBagConstraints.REMAINDER;
+        Button install = new Button("Install");
+        gridbag.setConstraints(install, c);
+        add(install);
+        install.addActionListener(e -> {
+            System.out.println("Installing...");
+            try {
+                if (this.loader == ModLoader.TWEAKER) {
+                    if (TweakerInstaller.installTweaker(Paths.get(path.getText()), this)) {
+                        System.out.println("Installation complete!");
+                        JOptionPane.showMessageDialog(this, "Installation complete!", "Success", JOptionPane.INFORMATION_MESSAGE);
+                        System.exit(0);
+                    }
+                } else {
+                    Path mods = Paths.get(path.getText());
+                    if (!Files.exists(mods)) {
+                            Files.createDirectory(mods);
+                    }
+                    Path file = Paths.get(BaritoneInstaller.class
+                        .getProtectionDomain()
+                        .getCodeSource()
+                        .getLocation()
+                        .toURI()
+                        .getPath()
+                    );
+                    Files.copy(file, mods.resolve(file.getFileName()), StandardCopyOption.REPLACE_EXISTING);
+                    System.out.println("Installation complete!");
+                    JOptionPane.showMessageDialog(this, "Installation complete!", "Success", JOptionPane.INFORMATION_MESSAGE);
+                    System.exit(0);
+                }
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                StringWriter writer = new StringWriter();
+                ex.printStackTrace(new PrintWriter(writer));
+                JOptionPane.showMessageDialog(this, "Error installing Baritone: " + ex.getMessage() + "\n\n" + writer, "Error", JOptionPane.ERROR_MESSAGE);
+            }
+        });
+    }
+
+
+    public String findMinecraftDir() {
+        if (OSProber.getOS() == OSProber.OS.WINDOWS) {
+            return "C:\\Users\\" + System.getProperty("user.name") + "\\AppData\\Roaming\\.minecraft";
```

Comment:
Wouldn't it be better to pull the %appdata% variable?

Musings:
I'm not sure that if someone moved their AppData folder to a different drive and created a Symbolic Link that the appdata variable will be updated as well, and I'm not sure if the File class will automatically follow a sym link.

Reply:
try to use user.home property instead of user.name. user.home gives you the same folder as doing %userprofile% in for example cmd would give.

Reply:
and when e.g. mincraft is installed on the root user(i know you shouldn't do this) it doesn't work becouse the user home of `root` is `/root/` not `/home/root/`